### ISSUE TITLE
UIAutomation2: Driver failed to start

### ISSUE THREAD

**Issue by amedvedjev** [appium/appium#7948]
## The problem

Failed to start driver with UIAutomation2. With old UIAutomation - no issues.

## Environment

* Appium version (or git revision) that exhibits the issue: 1.6.4-beta
* Last Appium version that did not exhibit the issue (if applicable): ?
* Mobile platform/version under test: Android
* Real device or emulator/simulator: Real device with Android 6.0

## Link to Appium logs
```
debug] [ADB] Running '/Users/Aleksei/Library/Android/sdk/platform-tools/adb' with args: ["-P",5037,"-s","LGH8155e9892e6","forward","--remove","tcp:8200"]
[MJSONWP] Encountered internal error running command: TypeError: Cannot read property 'isScreenLocked' of undefined
    at Object.callee$0$0$ (../../lib/android-helpers.js:419:18)
    at tryCatch (/usr/local/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:67:40)
    at GeneratorFunctionPrototype.invoke [as _invoke] (/usr/local/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:315:22)
    at GeneratorFunctionPrototype.prototype.(anonymous function) [as next] (/usr/local/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:100:21)
    at invoke (/usr/local/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:136:37)
    at enqueueResult (/usr/local/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:185:17)
    at F (/usr/local/lib/node_modules/appium/node_modules/core-js/library/modules/$.export.js:30:36)
    at AsyncIterator.enqueue (/usr/local/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:184:12)
    at AsyncIterator.prototype.(anonymous function) [as next] (/usr/local/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:100:21)
    at Object.runtime.async (/usr/local/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:209:12)
    at Object.callee$0$0 [as unlock] (../../lib/android-helpers.js:418:59)
    at AndroidUiautomator2Driver.startUiAutomator2Session$ (../../lib/driver.js:242:19)
    at tryCatch (/usr/local/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:67:40)
    at GeneratorFunctionPrototype.invoke [as _invoke] (/usr/local/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:315:22)
    at GeneratorFunctionPrototype.prototype.(anonymous function) [as next] (/usr/local/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:100:21)
    at GeneratorFunctionPrototype.invoke (/usr/local/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:136:37)
    at process._tickCallback (internal/process/next_tick.js:103:7)
[HTTP] <-- POST /wd/hub/session 500 7940 ms - 187
HTTP] --> POST /wd/hub/session {"capabilities":{"desiredCapabilities":{"app":"/Users/Aleksei/Downloads/test_clients/xxxx.apk","noReset":false,"clearSystemFiles":"true","appWaitActivity":"ee.xxx, ee.xxxxx, ee.xxxxx","disableAndroidWatchers":"true","version":"","deviceName":"LGH8155e9892e6","fullReset":true,"platform":"ANDROID","automationName":"UIAutomator2","browserName":"","autoGrantPermissions":"true","waitForAppScript":"$.delay(100); true;","udid":"LGH8155e9892e6","platformName":"Android"},"requiredCapabilities":{}}}
[debug] [MJSONWP] Bad parameters: BadParametersError: Parameters were incorrect. We wanted {"required":["desiredCapabilities"],"optional":["requiredCapabilities","capabilities","sessionId","id"]} and you sent ["capabilities"]
[HTTP] <-- POST /wd/hub/session 400 2 ms - 171 
System info: host: 'aleks.local', ip: '127.0.0.1', os.name: 'Mac OS X', os.arch: 'x86_64', os.version: '10.12.2', java.version: '1.8.0_101'
```

**Comment by mykola-mokhnach** [comment]
@jlipps @vrunoa FYI

**Comment by vrunoa** [comment]
@mykola-mokhnach looking

**Comment by stfcurry** [comment]
I found the same problem on the appium@1.6.3
**environment:** win10+real device(android 6.0.1)

**Comment by SrinivasanTarget** [comment]
@vrunoa Was there any update on this? I see this very often.

**Comment by vrunoa** [comment]
@SrinivasanTarget @mykola-mokhnach I made a change on appium-uiatomator2-driver with a fix.
https://github.com/appium/appium-uiautomator2-driver/pull/66

**Comment by amedvedjev** [comment]
fixed in 0.2.13
